pubpub/pubpub
# new: Google scholar citation_reference metatags

This generates Google/Highwire meta tags for references on the frontend for pubs that use built-in footnotes or citations. It's a pretty naive generator; on the other hand, the tags themselves are [not well specified](https://citation.js.org/blog/?post=3781582023912504297). Eventually, we should probably write a CSL to produce these, but that seems like a multi-week project not particularly worth our time at the moment.

Features of this particular PR:
- De-duplicates among footnotes and tags
- Supports journal, conference, and book (defaulting to journal)
- Filters out unstructured inputs

_Test plan_
1. Visit pub with lots of citations and verify they are represented in meta tags 
2. Visit pub with duplicate tags (both within footnote/citation and across footnote/citation) and verify they are _not_ replicated in meta tags. (ex: https://hdsr.duqduq.org/pub/hx3pgxjc)
3. Visit pub with journal, book, and conference citations and verify they are represented properly with `citation_journal_title`, `citation_book_title`, and `citation_conference_title` respectively.
4. Create new pub. Publish with no references. Then add some references and publish. Make sure meta tags are not there before, there after, and that merging works as expected.
